selfservice/strategy/password/login.go

```diff
@@ -68,6 +71,12 @@ func (s *Strategy) Login(w http.ResponseWriter, r *http.Request, f *login.Flow)
 		return nil, s.handleLoginError(w, r, f, &p, errors.WithStack(schema.NewInvalidCredentialsError()))
 	}
 
+	if !s.d.Hasher().IsSameAlgorithm([]byte(o.HashedPassword)) {
```

Comment:
~This check does not make sense here. The clear-text password does not contain the algorithm.~ - you fixed that already :wink: 

In the long term, we have to make it configurable which hash alg should be upgraded to which one. And some, like PBKDF2, will just be upgraded by default, as they are only intended for import.
For this PR it would be sufficient to just auto-upgrade PBKDF2 without a config option. So adding a check here that returns whether a hash is auto-upgrade (currently only PBKDF2), should be the way to go IMO.

Reply:
Made it configurable!

> One big thing missing now is documentation

Hmm... currently, Kratos doesn't have ability to import user credentials now.
So we have 2 choices to document currently.

* Document "Wait to resolve #605 " 323fdc3
* Implement for #605 in thi PR

Which do you think better?